EdvClient.js

```diff
@@ -352,29 +352,50 @@ export class EdvClient {
    *   capability to use to authorize the invocation of this operation.
    * @param {Object} options.invocationSigner - An API with an
    *   `id` property and a `sign` function for signing a capability invocation.
+   * @param {function} [keyResolver=this.keyResolver] a function that returns
+   *   a Promise that resolves a key ID to a DH public key.
    *
    * @return {Promise<Boolean>} resolves to `true` if the document was deleted
    *   and `false` if it did not exist.
    */
-  async delete({id, capability, invocationSigner}) {
+  async delete({id, capability, invocationSigner,
+    keyResolver = this.keyResolver}) {
     _assertString(id, '"id" must be a string.');
     _assertInvocationSigner(invocationSigner);
 
+    let doc;
+    try {
+      doc = await this.get({id, capability, invocationSigner});
+    } catch(e) {
+      if(e.message === 'Document not found.') {
+        return false;
+      }
+      throw e;
+    }
```

Comment:
I don't think we can safely do this -- it creates race conditions where the user of the client may need to depend on the sequence number to perform their delete. We should instead have the same parameters as the `update` call -- where the user has to pass in the doc themselves. We also need to allow for a keyAgreement key to be passed in (vs. only allowing a default -- if even set -- to be used). In short, we need the parameters to be pretty much the same as `update` -- but, internally, we will make a shallow copy of `doc` and then clear content, meta, stream, and set `meta.deleted = true` before passing `doc` to `update`.

I haven't thought about how to inform the server that certain chunks for the `stream` (if any was associated with the deleted document) can be removed. We can punt on that for now (unless you have some ideas around what could/should be done). I suspect we're going to have to surface the total number of chunks to the server at some point, but we haven't considered exactly how to do this properly yet. Just food for thought if you're interested in EDV design! :)